api/filter_dimension_options_test.go

```diff
@@ -397,6 +397,45 @@ func TestSuccessfulPatchFilterBlueprintDimension(t *testing.T) {
 		So(w.Code, ShouldEqual, http.StatusOK)
 	})
 
+	Convey("Successfully patch dimension options, with a single valid 'remove' patch operation with a mix of existent and inexistent options, returns 200", t, func() {
+		reader := strings.NewReader(`[
+			{"op":"remove", "path": "/options/-", "value": ["33", "123456789"]}
+		]`)
+		r, err := http.NewRequest("PATCH", "http://localhost:22100/filters/12345678/dimensions/age", reader)
+		So(err, ShouldBeNil)
+
+		w := httptest.NewRecorder()
+		api := Setup(host, maxRequestOptions, mux.NewRouter(), &mocks.DataStore{}, &mocks.FilterJob{}, &mocks.DatasetAPI{}, previewMock, enablePrivateEndpoints, downloadServiceURL, downloadServiceToken, serviceAuthToken)
+		api.router.ServeHTTP(w, r)
+		So(w.Code, ShouldEqual, http.StatusOK)
```

Comment:
Should this test also be asserting the expected remove was carried out?

Reply:
Yes, I found that the mocks are not create through `moq`, they are simply an implementation of the interface, covering different cases. I started to try to make the mocks extensions of the `moq` mock, but it was quite a lot of changes. I can continue that work, it will also be useful for other tests.